test/chain-tree-compaction-test.js

```diff
@@ -113,15 +114,43 @@ describe('Tree Compacting', function() {
         }
       }
 
-      let name, nameHash;
-      const treeRoots = [];
+      let name, nameHash, listener;
+      let treeRoots = [];
+
+      const checkTree = async (tree, compacted = false) => {
+        for (const [index, hash] of treeRoots.entries()) {
+          if (compacted && index < (treeRoots.length - 8)) {
```

Comment:
We expect to always have last 8 tree roots because in the beginning of the test we make sure to have `1:8` ratio with `treeInterval` and `keepBlocks`. In the beginning of the test we mine `20 blocks` (maybe that should be dynamic based on those)